code/src/components/Form.js

```diff
@@ -0,0 +1,305 @@
+import React, { useState } from 'react';
+
+import InputText from './InputText';
+import InputSelect from './InputSelect';
+import InputCheckbox from './InputCheckbox';
+import InputRadiobutton from './InputRadiobutton';
+import Summary from './Summary';
+import Button from './Button';
+import ProgressBar from './ProgressBar';
+import '../styles/Form.scss';
+
+const Form = () => {
+  const [pageIndex, setPageIndex] = useState(-1);
+  const [text, setText] = useState('');
+  const [select, setSelect] = useState('');
+  const [radiobutton, setRadio] = useState();
+  const [checkbox, setCheckbox] = useState([]);
+  const [submit, setSubmit] = useState(false);
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    setSubmit(true);
+  };
+  const handleTextChange = event => {
+    setText(event.target.value);
+  };
+  const handleSelectChange = event => {
+    setSelect(event.target.value);
+  };
+  const handleRadioChange = event => {
+    setRadio(event.target.value);
+  };
+  const handleCheckboxChange = checkboxValue => {
+    checkbox.includes(checkboxValue)
+      ? setCheckbox(checkbox.filter(item => item !== checkboxValue))
+      : setCheckbox([...checkbox, checkboxValue]);
+  };
+
+  const onNextPageChange = () => {
+    setPageIndex(pageIndex + 1);
+  };
+  const onPreviousPageChange = () => {
+    setPageIndex(pageIndex - 1);
+  };
+  const onRefresh = () => {
+    window.location.reload();
+  };
+
+  const selectArray = [
+    {
+      text: 'Select...',
+      value: '',
+    },
+    {
+      text: '1-10',
+      value: '1-10',
+    },
+    {
+      text: '11-20',
+      value: '11-20',
+    },
+    {
+      text: '21-30',
+      value: '21-30',
+    },
+  ];
+  const checkboxArray = [
+    {
+      text: '1-10',
+      value: ['1-3', '4-7', '8-10'],
+    },
+    {
+      text: '11-20',
+      value: ['11-13', '14-17', '18-20'],
+    },
+    {
+      text: '21-30',
+      value: ['21-23', '24-27', '28-30'],
+    },
+  ];
+  const questions = {
+    text: 'What is the text input?',
+    select: 'What is the select?',
+    checkbox: 'What is the checkbox?',
+    radio: 'What is the radiobutton?',
+  };
+
+  return (
+    <main className="App__main">
+      {!submit ? (
+        <form className="Form" onSubmit={handleSubmit}>
+          <h1 className="App__header">Excellent survey</h1>
+
+          {pageIndex === -1 && (
+            <section className="Form__section">
+              <h2 className="App__subheader">
```

Comment:
Appreciate BEM - in your case as always - but we should stick to lowercase naming :

```
<section className="form__section">
    <h2 className="app__subheader">
```